sdcm/gce_log_monitor.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import time
+from typing import Dict
+
+from dateutil import parser
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+from sdcm.keystore import KeyStore
+from sdcm.sct_events import Severity
+from sdcm.sct_events.base import SctEvent, EventPeriod
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GceLoggingClient:  # pylint: disable=too-few-public-methods
+
+    def __init__(self):
+        raw_credentials = KeyStore().get_logging_gcp_credentials()
+        self.credentials = service_account.Credentials.from_service_account_info(raw_credentials)
+        self.project_id = raw_credentials['project_id']
+
+    def get_error_logs(self, since: float):
+        """Gets 'hostError' and 'automaticRestart' logs entries from GCE since the time provided in argument.
+        Returns list of entries in a form of dictionaries. See example output in unit tests."""
+        since = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime(since))
+        with build('logging', 'v2', credentials=self.credentials, cache_discovery=False) as service:
+            return self._get_log_entries(service, since)
+
+    def _get_log_entries(self, service, since, page_token=None):
+        body = {
+            "resourceNames": [
+                f"projects/{self.project_id}"
+            ],
+            "filter": f'protoPayload.methodName="compute.instances.automaticRestart" '
+                      f'OR protoPayload.methodName="compute.instances.hostError" timestamp > "{since}"'
+        }
+        if page_token:
+            body.update({"page_token": page_token})
+        ret = service.entries().list(body=body).execute()
+        entries = ret.get('entries', [])
+        if page_token := ret.get("nextPageToken"):
+            entries.extend(self._get_log_entries(service, since, page_token))
+        return entries
+
+
+class GceInstanceEvent(SctEvent):
```

Comment:
I think we should move this one to be under the event module

Reply:
I don't see the benefits of that. Currently we have everything related in one place and I don't see it could be used anywhere else.